extra/release_info_generator.py

```diff
@@ -0,0 +1,131 @@
+LtsEnd = namedtuple("LtsEnd", ["isExpired", "date"])
+
+releaseMatcher = compile(r"([0-9]+)\.([0-9]+)\.([0-9]+)")
+tagMatcher = compile(r"tag:'(.*)' datetime:'(\d+)'")
+
+lastChecked = getenv("LAST_CHECKED", "2.2.0")
+endOfToday = datetime.combine(datetime.now(), time.max)
+
+
+def determine_lts_end(tag, currentMinor, releases):
+    match = releaseMatcher.match(tag.version)
+    # for now we assume all even releases are LTS
+    if int(match.group(2)) % 2:
+        return
+    aYear = timedelta(
+        days=366
+        if (
+            (tag.datetime.month >= 3 and isleap(tag.datetime.year + 1))
+            or (tag.datetime.month < 3 and isleap(tag.datetime.year))
+        )
+        else 365
+    )
+    expirationDate = tag.datetime + aYear
+    if currentMinor in releases:
+        eolDate = releases[currentMinor]["supported_until"].split("-")
+        expirationDate = datetime.fromisoformat(
+            f"{eolDate[0]}-{eolDate[1]}-{monthrange(int(eolDate[0]), int(eolDate[1]))[1]}T{str(time.max)}"
+        )
+    isExpired = expirationDate < endOfToday
+    return LtsEnd(isExpired, expirationDate)
+
+
+def collect_release_info(tag, minorRelease):
+    args = {
+        "list": "git",
+        "list_format": "json",
+        "in_integration_version": tag.version,
+    }
+    repoList = StringIO()
+    with redirect_stdout(repoList):
+        do_list_repos(
+            Namespace(**args),
+            False,
+            False,
+            False,
+        )
+
+    versionReposInfo = json.loads(repoList.getvalue())
+    existingInfo = {}
+    if tag.version in minorRelease:
+        existingInfo = minorRelease[tag.version]
+    result = {
+        "release_date": tag.datetime.strftime("%Y-%m-%d"),
+        **existingInfo,
+        **versionReposInfo,
+    }
+    result["repos"] = list(filter(lambda repo: (repo["version"]), result["repos"]))
+    return result
+
+
+def get_releases():
+    tags = []
+    saasTags = []
+    with open("tags", "r") as tagsFile:
+        for line in tagsFile:
+            tagMatch = tagMatcher.match(line.strip())
+            if not tagMatch:
+                continue
+            tag = Tag(tagMatch.group(1), datetime.fromtimestamp(int(tagMatch.group(2))))
+            if tag.version.startswith("saas"):
+                saasTags.append(
+                    {"tag": tag.version, "date": tag.datetime.strftime("%Y-%m-%d")}
+                )
+            match = releaseMatcher.fullmatch(tag.version)
+            if match and tag.version >= lastChecked:
+                tags.append(tag)
+    return {"saasTags": saasTags, "tags": tags}
+
+
+taggedReleases = get_releases()
+releases = taggedReleases["tags"]
+releaseInformation = {"lts": [], "releases": {}, "saas": taggedReleases["saasTags"]}
+
+existingReleaseInformation = {}
+try:
+    with open("versions.json", "r") as current:
+        existingReleaseInformation = json.load(current)
+except:
+    existingReleaseInformation = existingReleaseInformation
```

Comment:
Nitpick: Isn't this really a no-op, essentially `pass`?

Reply:
ah yeah, it's been a while I `python`ed - I forgot about `pass` 🙈 - adjusted!